### PR TITLE
mbeliaev/refactoring

### PR DESCRIPTION
**Pull request by beliaev-maksim**
some minor refactoring, fix types, doc strings, etc

PyCharm identified issues on latest master:
image

PyCharm identified issues after merging this PR:
image

### CODE REVIEW

File: responses/__init__.py [hunk 1]
```diff
@@ -570,6 +570,8 @@ def __init__(
         self.response_callback = response_callback
         self.passthru_prefixes = tuple(passthru_prefixes)
         self.target = target
+        self._patcher = None
+        self._matches = []
```
Comment (markstory) [inline responses/__init__.py:574]: Doesn't the reset() call on line 568 do this?

Comment (beliaev-maksim) [reply responses/__init__.py:574]: @markstory 
yes, it does, but by python convention all class attributes should be initialized in `__init__` method

Comment (markstory) [reply responses/__init__.py:574]: Ok. Couldn't we remove the call to `reset()` then?

Comment (beliaev-maksim) [reply responses/__init__.py:574]: @markstory 
`reset` also resets calls:
`self._calls.reset()`

**Pull request merged**